x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/services/cohere/CohereServiceTests.java

```diff
@@ -0,0 +1,853 @@
+                "id",
+                TaskType.TEXT_EMBEDDING,
+                persistedConfig.config(),
+                persistedConfig.secrets()
+            );
+
+            MatcherAssert.assertThat(model, instanceOf(CohereEmbeddingsModel.class));
+
+            var embeddingsModel = (CohereEmbeddingsModel) model;
+            MatcherAssert.assertThat(embeddingsModel.getServiceSettings().uri().toString(), is("url"));
+            MatcherAssert.assertThat(
+                embeddingsModel.getTaskSettings(),
+                is(new CohereEmbeddingsTaskSettings("model", InputType.SEARCH, null, null))
+            );
+            MatcherAssert.assertThat(embeddingsModel.getSecretSettings().apiKey().toString(), is("secret"));
+        }
+    }
+
+    public void testParsePersistedConfig_CreatesACohereEmbeddingsModel() throws IOException {
+        try (
+            var service = new CohereService(
+                new SetOnce<>(mock(HttpRequestSenderFactory.class)),
+                new SetOnce<>(createWithEmptySettings(threadPool))
+            )
+        ) {
+            var persistedConfig = getPersistedConfigMap(
+                getServiceSettingsMap("url"),
+                getTaskSettingsMap("model", null, null, CohereTruncation.NONE)
+            );
+
+            var model = service.parsePersistedConfig("id", TaskType.TEXT_EMBEDDING, persistedConfig.config());
+
+            MatcherAssert.assertThat(model, instanceOf(CohereEmbeddingsModel.class));
+
+            var embeddingsModel = (CohereEmbeddingsModel) model;
+            MatcherAssert.assertThat(embeddingsModel.getServiceSettings().uri().toString(), is("url"));
+            MatcherAssert.assertThat(
+                embeddingsModel.getTaskSettings(),
+                is(new CohereEmbeddingsTaskSettings("model", null, null, CohereTruncation.NONE))
+            );
+            assertNull(embeddingsModel.getSecretSettings());
+        }
+    }
+
+    public void testParsePersistedConfig_ThrowsErrorTryingToParseInvalidModel() throws IOException {
+        try (
+            var service = new CohereService(
+                new SetOnce<>(mock(HttpRequestSenderFactory.class)),
+                new SetOnce<>(createWithEmptySettings(threadPool))
+            )
+        ) {
+            var persistedConfig = getPersistedConfigMap(getServiceSettingsMap("url"), getTaskSettingsMapEmpty());
+
+            var thrownException = expectThrows(
+                ElasticsearchStatusException.class,
+                () -> service.parsePersistedConfig("id", TaskType.SPARSE_EMBEDDING, persistedConfig.config())
+            );
+
+            MatcherAssert.assertThat(
+                thrownException.getMessage(),
+                is("Failed to parse stored model [id] for [cohere] service, please delete and add the service again")
+            );
+        }
+    }
+
+    public void testParsePersistedConfig_CreatesAnCohereEmbeddingsModelWithoutUrl() throws IOException {
```

Comment:
nit: would be nice to replace "AnCohere" with "ACohere" to match the rest of the methods in this file.